lib/gitsh/git_repository.rb

```diff
@@ -118,6 +123,21 @@ def abbreviated_sha
       end
     end
 
+    def modern_git_available_config_variables
+      command = git_command("config --list --name-only")
+      out, _, status = Open3.capture3(command)
+
+      if status.success?
+        out.chomp.lines.map { |line| line.chomp.to_sym }
+      end
+    end
+
+    def old_git_available_config_variables
+      git_output('config --list').lines.map do |line|
+        line.split('=').first.to_sym
```

Comment:
Hmm...given that `#split` still returns an array in that case, it's unclear whether that's improving things much. The best alternative I could come up with was

```
line.each_char.take_while { |c| c != "-" }.join
```

But given that config lines aren't usually very long (I assume), it's probably not a big deal either way.